torchtune/modules/transforms/vision_utils/get_resize_dimensions.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+
+import math
+
+from typing import Optional, Tuple
+
+logger = logging.getLogger(__name__)
+
+
+def get_resize_dimensions(
+    image_size: Tuple[int], target_size: Tuple[int], max_upscaling_size: Optional[int]
+) -> Tuple[int]:
+    """
+    Calculates the resize dimensions for an image, given the target size and max_upscaling_size.
```

Comment:
I think this could be fleshed out a bit more. The key thing this function does is computes what size an image would be if it was resized to be inscribed within a target size. So it's upscaled or downscaled such that one size equals the target size and the second side is less than or equal to the target size. I think some description describing this would help make the function easier to discover and use.

Reply:
thanks, updated